### ISSUE TITLE
Compilation error if using UVM without UVM report mock API

### ISSUE THREAD

**Issue by MarcinRo** [svunit/svunit#42]
It seems the _issue_ I reported [on Sourceforge](https://sourceforge.net/p/svunit/tickets/72/) has never been really dealt with, so I am **resubmitting** it.

While trying to compile my unit test for a UVM class, the SV unit auto-generated classes contain imports of UVM report mock API although I do not use it.

`file: .testrunner.sv`
`  import svunit_uvm_mock_pkg::svunit_uvm_test_inst;`
`ncvlog: *E,NOPBIND (.testrunner.sv,8|27): Package svunit_uvm_mock_pkg could not be bound.`
`  import svunit_uvm_mock_pkg::uvm_report_mock;`
`ncvlog: *E,NOPBIND (.testrunner.sv,9|27): Package svunit_uvm_mock_pkg could not be bound.`

My [fix](https://sourceforge.net/p/svunit/tickets/72/attachment/Screenshot%202015-08-07%2012.17.47.png) proposed on Sourceforge still works for me although it is a bit nasty I have to locally edit my copy of _bin\create_testrunner.pl_ file.

**Comment by MarcinRo** [comment]
Not sure if #37 will fix it. Probably not.

**Comment by MarcinRo** [comment]
OK, I ran into other problems with this fix (a fatal error at runtime). It looks like I misunderstood the meaning of RUN_SVUNIT_WITH_UVM_REPORT_MOCK. Whether the svunit_uvm_mock_pkg is imported or not should be ruled by RUN_SVUNIT_WITH_UVM. RUN_SVUNIT_WITH_UVM_REPORT_MOCK causes a callback to be added, which then causes errors raised by the component under test to be suppressed.

I think it is worth adding a fix suggested by @tudortimi under #9. Closing this ticket.
